frontend/src/api/mockAPI/sessions.js

```diff
@@ -26,6 +26,28 @@ export class Session extends MockAPIController {
             throw new Error(message);
         }
     }
+
+    validateProp(obj, prop) {
+        if (prop === "name") {
+            if (obj[prop] === undefined || obj[prop].length === 0) {
+                throw new Error(
+                    `Property ${prop} cannot be empty or undefined.`
+                );
+            } else {
```

Comment:
I don't understand what you're doing. I suspect it's wrong (i.e., it doesn't belong in this function). You need comments to explain.

Reply:
say we want to update a session. `obj` here is the request body `{ id: 1, name: xxx}`. `prop` is the key/field. if the field is `name` , we need to make sure `{ ...,name: ""}` and `{..., name: undefined}` throws an error. And if `name` is not empty, we still need to check if it's updatable, ie, if it is unique to the rest of the sessions (all the session that id is not 1)

Reply:
You should be passing in the prop and the value, not the prop and the object.

Reply:
Hmm... You mean it should throw errors but return false or a message?